File: YEAR2/SEM2/Medii de proiectare si programare/ConsoleApp1/repository0/ProbaDBRepo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ConsoleApp1.domain;
using ConsoleApp1.repository;
using log4net;
using System.Data;

namespace ConsoleApp1.repository0
{
    public class ProbaDBRepo : IProba
    {
        private static readonly ILog log = LogManager.GetLogger("ProbaDBRepo");

        IDictionary<String, string> props;

        public ProbaDBRepo(IDictionary<String, string> props)
        {
            log.Info("Creating ProbaDBRepo");
            this.props = props;
        }
        public Proba Delete(long id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Proba> FindAll()
        {
            throw new NotImplementedException();
        }

        public Proba FindOne(long id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection(props);
            Proba proba = null;
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select varstamin,varstamax,proba from Proba  where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {

                        int varstamin = dataR.GetInt32(0);
                        int varstamax = dataR.GetInt32(1);
                        string proba1 = dataR.GetString(2);
                        proba = new Proba(varstamin, varstamax,proba1);
                        proba.id = id;
                        log.InfoFormat("Exiting findOne with value {0}", proba);
                        return proba;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", proba);
            return proba;
        }

  

    public Proba Save(Proba entity)
        {
            throw new NotImplementedException();
        }

    public Proba Update(Proba entity)
        {
           throw new NotImplementedException();
        }
    }
}

File: YEAR2/SEM2/Medii de proiectare si programare/Tema1/src/main/java/org/example/repository/database/ProbaDBRepo.java
package org.example.repository.database;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.example.domain.Proba;
import org.example.repository.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class ProbaDBRepo implements IProba{

    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public ProbaDBRepo(Properties props){
        logger.info("Initializing ProbaDBRepo with properties:{}",props);
        dbUtils = new JdbcUtils(props);
    }
    @Override
    public Proba findOne(Long aLong) {
        logger.traceEntry("findone task{}, elem");
        Connection connection = dbUtils.getConnection();
        Proba proba = null;
        try (PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM Proba WHERE id=(?)")) {
            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                int varstamin = resultSet.getInt("varstamin");
                int varstamax = resultSet.getInt("varstamax");
                String proba1 = resultSet.getString("proba");
                proba = new Proba(varstamin, varstamax, proba1);
                proba.setId(aLong);
            }

        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB" + e);
        }
        logger.traceExit(proba);
        return proba;
    }

    @Override
    public Iterable<Proba> findAll() {
        return null;
    }

    @Override
    public Proba save(Proba entity) {
        return null;
    }

    @Override
    public Proba delete(Long aLong) {
        return null;
    }

    @Override
    public Proba update(Proba entity) {
        return null;
    }
}
